preprocessor.py

```diff
@@ -75,32 +93,64 @@ def tag_usernames(text):
 
 
 def preprocess(
-    df, emojis, emoticons, urls, html_tags, acronyms, spelling, usernames,
+    in_filename,
+    out_filename,
+    emojis,
+    emoticons,
+    urls,
+    html_tags,
+    acronyms,
+    spelling,
+    usernames,
+    max_workers,
+    chunksize,
 ):
-    if emojis:
-        print("Removing emojis...")
-        df["text"] = df["text"].progress_apply(lambda text: remove_emojis(text))
-    if emoticons:
-        print("Removing emoticons...")
-        df["text"] = df["text"].progress_apply(lambda text: remove_emoticons(text))
+    df = pd.read_csv(in_filename)
+
+    if chunksize == -1:
+        if df.shape[0] < 1000:
+            chunksize = 1
+        else:
+            chunksize = min(df.shape[0] // max_workers, 1000)
+
     if html_tags:
         print("Removing html tags...")
-        df["text"] = df["text"].progress_apply(lambda text: remove_html_tags(text))
+        df["text"] = process_map(
+            remove_html_tags, df["text"], max_workers=max_workers, chunksize=chunksize
+        )
     if urls:
         print("Tagging URLs...")
-        df["text"] = df["text"].progress_apply(lambda text: tag_urls(text))
+        df["text"] = process_map(
+            tag_urls, df["text"], max_workers=max_workers, chunksize=chunksize
+        )
     if usernames:
         print("Tagging usernames...")
-        df["text"] = df["text"].progress_apply(lambda text: tag_usernames(text))
+        df["text"] = process_map(
+            tag_usernames, df["text"], max_workers=max_workers, chunksize=chunksize
+        )
     if acronyms:
         print("Converting acronyms...")
-        df["text"] = df["text"].progress_apply(lambda text: convert_acronyms(text))
+        df["text"] = process_map(
+            convert_acronyms, df["text"], max_workers=max_workers, chunksize=chunksize
+        )
+    if emojis:
+        print("Removing emojis...")
+        df["text"] = process_map(
+            remove_emojis, df["text"], max_workers=max_workers, chunksize=chunksize
+        )
+    if emoticons:
+        print("Removing emoticons...")
+        df["text"] = process_map(
+            remove_emoticons, df["text"], max_workers=max_workers, chunksize=chunksize
+        )
     if spelling:
-        print("Spellchecking...")
-        df["text"] = df["text"].progress_apply(lambda text: correct_spelling(text))
+        print("Spellchecking (it takes a while)...")

```

Comment:
For large chunk sizes, the progress bar might take a while to update, since it updates in chunks/steps.